docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,85 @@
+        * This replaces our current release process -- no more cutting releases.
+        * This means that our version numbers will no longer follow semver.  Breaking changes will not cause the major version to be incremented, we will only do that on merge day.
+        * To push changes to a consumer: create a PR in their repository that bumps the application-services version to an existing release.
+        * All of these changes can be done with little impact to the application teams.  From their POV, the only change is that we have a new versioning system.
+    * Phase 2: Automate PRs to merge application-services code into consumer applications
+      * Whenever a new application-services version is released (including nightly, beta, release, and esrs)
+        * For each consumer application (including all mobile apps and Desktop):
+            * Create a PR to take the new application-services version.
+            * Set up an auto-merge system where if the test suite passes we automatically merge the PR.
+            * Set up an alert that gets sent to `sync-bots` if the test suite fails.  This means we would need to identify why the tests failed and assign a developer to resolve the issue.  The expectation would be that the developer fixes things in time for the next nightly merge.
+    * In order to keep things running smoothly, if a breaking change requires serious review from the consumer teams, the PRs for that change should be approved and ready to merge before that breaking change is merged into application-services.  If we accidentally merge the change into application-services before the PRs are ready, then we should back the change out.
+    * On the monthly Firefox merge date, roll each branch forward:
+      - Create a new `esr-[version]` branch from `release`, if needed
+      - Merge `beta` into `release`
+      - Merge `main` into `beta`
+      - Bump the major version number in main and start a new changelog header.
+    * Coordinate this all with release engineering and see how they want to be involved.
+
+## Decision Outcome
+
+**(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+
+## Pros and Cons of the Options
+
+### Keep our current release process
+
+  * Good, because it doesn't require changes to our workflow
+
+### Move to a rapid release process where changes in app-services are frequently merged into consumer applications
+
+  * Good, because code will get merged and released more frequently.
+  * Good, because we don't need to spend time cutting releases.
+  * Good, because uplifts can be done much quicker.
+  * Bad, because we need implement new automation.
+  * Bad, because when breaking changes cause the automatic nightly merge to fail for an application.  This means a developer will get the relatively urgent task of resolving things.
```

Comment:
Maybe it's better to just back out breaking changes when they happen? The idea that quickly fixing a breaking change (which could be nontrivial) might fall on whoever is around seems undesirable.

Reply:
Yeah, that sounds better to me as well.